sycl/plugins/hip/CMakeLists.txt

```diff
@@ -122,6 +122,7 @@ add_sycl_plugin(hip
     ${sycl_plugin_dir}
     ${CMAKE_CURRENT_SOURCE_DIR}/../unified_runtime
     ${UNIFIED_RUNTIME_SOURCE_DIR}/source/ # for adapters/hip
+    "${SYCL_SOURCE_DIR}/source" # for compiler.hpp
```

Comment:
`pi2ur` depends on compiler interfaces and all plugins depend on `pi2ur`. Therefore, this is a necessary evil (?) for now.
My assumption is that this include won't be needed once we drop plugins in favor of UR.